site/source/docs/porting/simd.rst

```diff
@@ -35,6 +35,45 @@ When porting native SIMD code, it should be noted that because of portability co
 
 SIMD-related bug reports are tracked in the `Emscripten bug tracker with the label SIMD <https://github.com/emscripten-core/emscripten/issues?q=is%3Aopen+is%3Aissue+label%3ASIMD>`_.
 
+===========================
+Optimization considerations
+===========================
+
+When porting SIMD code to use WebAssembly SIMD, implementors should be aware of semantic differences between the host hardware and WebAssembly semantics; as acknowledged in the WebAssembly design documentation, "`this sometimes will lead to poor performance <https://github.com/WebAssembly/design/blob/master/Portability.md#assumptions-for-efficient-execution>`." The following list outlines some WebAssembly SIMD instructions to look out for when performance tuning:
+
+.. list-table:: WebAssembly SIMD instructions with performance implications
+   :widths: 10 10 30
+   :header-rows: 1
+
+   * - WebAssembly SIMD instruction
+     - Hardware architecture
+     - Considerations
+
+   * - [f32x4|f64x2].[min|max]
+   * - x86
+   * - As with the scalar versions, the NaN propagation semantics force runtimes to emulate with 8+ x86 instructions; if possible, use [f32x4|f64x2].[pmin|pmax] instead (1 x86 instruction).
+
+   * - i32x4.trunc_sat_f64x2_[u|s]_zero
+   * - x86
+   * - No equivalent x86 semantics; emulated with 5-6 x86 instructions.
+
+   * - f32x4.convert_i32x4_[s|u]
+   * - x86
+   * - No equivalent x86 semantics; emulated with 8-14 x86 instructions.
+
+   * - i8x16.[shl|shr_s|shr_u]
+   * - x86
+   * - Included for orthogonality, these instructions have no equivalent x86 instruction and are emulated with 5-11 x86 instructions (i.e. using 16x8 shifts).
```

Comment:
perhaps a comment saying shifts by a constant is special cased by engines and lead to better codegen (applies to all shifts)

Reply:
Added a new row for all shifts...